### ISSUE TITLE
TRX report shows test as failed when passed in rerun

### ISSUE THREAD

**Issue by zelda1link3** [microsoft/testfx#2380]
When using mstestrunner with the TRX report plugin and the retry plugin. When a test fails the first run and succeeds the second run. It all sees it as 1 big run and not a rerun in Azure devops.
I expect that it would show as passed with reruns and not failed like it does now.

**Comment by MarcoRossignoli** [comment]
The retry extensions generates a new trx for every run inside the attempt folder and will move "the last" one trx to the root of the "result directory". The reason is why in case of failure you're usually interested only to the failed tests. 
You can also inspect all the others trx generated in the attempts folders(the directory is created under TestResult/Retries/Guid).


> It all sees it as 1 big run and not a rerun in Azure devops.

Retry extensions is a portable retry feature build for any CI systems and doesn't know Azure devops. 

The ux logic of Azure devops in case of multiple trx with different attempt in it is out of the scope of the extension.

I'll reach out the Azure devops team to understand if they can accomodate  this extension.

**Comment by zelda1link3** [comment]
We use --report-trx-filename and we only read that trx file. in this trx file all the runs are recorded.

**Comment by MarcoRossignoli** [comment]
> We use --report-trx-filename and we only read that trx file. in this trx file all the runs are recorded.

Are you using the Retry extension https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-runner-extensions#retry ?

**Comment by MarcoRossignoli** [comment]
Can you check what AzDo is "uploading"? Maybe the filter is `/**/*.trx` and so AzDo takes the "first" attempt file only and not the last one inside the final test result directory.

**Comment by zelda1link3** [comment]
Yes we are using the retry extension. We see that there are multiple trx files in azure devops when we download the zip. But we use **/TEST_PROJECTNAME.trx.  so the same file we put in the --report-trx-filename. 
the complete command is:

.\$(TestProjectName).exe --report-trx --report-trx-filename TEST_$(TestProjectName).trx --retry-failed-tests 3 --retry-failed-tests-max-percentage $(PercentageRetryTests)

**Comment by zelda1link3** [comment]
Okay thanks that is the issue for now.

**Comment by MarcoRossignoli** [comment]
I'll keep this issue open to track AzDo/Retry integration

**Comment by Evangelink** [comment]
I'll move forward by closing this issue. Please feel free to comment if there was something else not covered here.
